metricproviders/webmetric/webmetric.go

```diff
@@ -0,0 +1,229 @@
+package webmetric
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/utils/evaluate"
+	metricutil "github.com/argoproj/argo-rollouts/utils/metric"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+const (
+	//ProviderType indicates the provider is prometheus
+	ProviderType = "WebMetric"
+)
+
+// Provider contains all the required components to run a WebMetric query
+// Implements the Provider Interface
+type Provider struct {
+	logCtx     log.Entry
+	client     *http.Client
+	jsonParser *jsonpath.JSONPath
+}
+
+// Type incidates provider is a WebMetric provider
+func (p *Provider) Type() string {
+	return ProviderType
+}
+
+func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alpha1.Measurement {
+	var (
+		err error
+	)
+
+	startTime := metav1.Now()
+
+	// Measurement to pass back
+	measurement := v1alpha1.Measurement{
+		StartedAt: &startTime,
+	}
+
+	// Create request
+	request := &http.Request{
+		Method: "GET", // TODO maybe make this configurable....also implies we will need body templates
+	}
+
+	request.URL, err = url.Parse(metric.Provider.Web.URL)
```

Comment:
This one I actually did since we can't have a ':=' assignment combined with the field  assignment I have there. I could swap that out with a new var that then gets assigned to request.URL, I'll push that in the next commit, and we can nitpick that if it's a problem also.